storage/innobase/xtrabackup/src/write_filt.cc

```diff
@@ -107,12 +108,22 @@ static bool wf_incremental_process(xb_write_filt_ctxt_t *ctxt,
   ulint page_size = cursor->page_size;
   byte *page;
   xb_wf_incremental_ctxt_t *cp = &(ctxt->wf_incremental_ctxt);
-
   for (i = 0, page = cursor->buf; i < cursor->buf_npages;
        i++, page += page_size) {
-    if (incremental_lsn > mach_read_from_8(page + FIL_PAGE_LSN)) {
+    /*
+     * metadata_from_lsn stores LSN of start of previous backup, while
+     * incremental_lsn stores the last checkpoint at the end of previous backup.
+     * We use metadata_from_lsn for mysql.ibd because we skip applying
+     * MLOG_TABLE_DYNAMIC_META events if we are not preparing the last
+     * incremental backup. In order to capture changes that happened to
```

Comment:
looks good mostly. Improved a bit. Let me know if this is OK?

We use metadata_from_lsn for mysql.ibd because we skip applying logical redos (MLOG_TABLE_DYNAMIC_META) during the incremental prepare (except the last prepare). These logical redos are converted to regular redo and flushed to pages in mysql.ibd. So we directly take the physical changes made to innodb_dynamic_metadata since the last backup. Hence we copy all mysql.ibd since last backup */

Reply:
@satya-bodapati 

>  These logical redos are converted to regular redo and flushed to pages in mysql.ibd
I would enhanced this one to:

 These logical redos are converted to regular redo and flushed to pages in mysql.ibd when the server process a checkpoint.

Reply:
Agree, Looks good.